src/middlewared/middlewared/plugins/interface/vrrp_linux.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding=utf8 -*-
+import json
+import subprocess
+
+from middlewared.service import Service
+
+
+class VrrpService(Service):
+
+    class Config:
+        private = True
+        namespace_aliases = 'interfaces'
+
+    def vrrp_config(self, ifname):
+
+        # query db for configured settings
+        info = self.middleware.call_sync(
+            'datastore.query',
+            'network.interfaces',
+            [('int_interface', '=', ifname)],
+        )
+        configured_vips = [
+            i['int_vip'] for i in info if i['int_vip']
+        ]
+
+        # if there are no VIPs then there is no reason to continue
+        if not configured_vips:
+            self.logger.warning(
```

Comment:
For all logging operations we should use %s/%r/%s instead of f-string, its useful to for logging to not receive the already formatted string because it can aggregate on same errors messages in the future.